pptxHandler.py

```diff
@@ -0,0 +1,145 @@
+MAX_RETRIES = 3  # Maximum number of retries for slide processing
+
+
+class SlideProcessingError(Exception):
+    pass
+
+
+class PresentationProcessingError(Exception):
+    pass
+
+
+class OpenAIError(Exception):
+    pass
+
+
+async def process_slide(slide_num, slide_text, retry_count=1):
+    """
+    Processes a slide by generating an explanation using OpenAI's ChatCompletion model.
+
+    Args:
+        slide_num (int): The slide number.
+        slide_text (str): The text content of the slide.
+        retry_count (int): The number of times the slide processing has been retried (default: 1).
+
+    Returns:
+        str: The generated explanation for the slide.
+
+    Raises:
+        SlideProcessingError: If the slide processing fails after the maximum number of retries.
+    """
+    try:
+        print(f"Processing slide {slide_num}...\n")
+        prompt = f"{PROMPT_INIT}\n{slide_text}\n"
+        response = await openai.ChatCompletion.acreate(
+            model="gpt-3.5-turbo",
+            messages=[
+                {"role": "system", "content": "You are a helpful assistant that explains PowerPoint slides:"},
+                {"role": "user", "content": prompt}
+            ]
+        )
+        explanation = response.choices[0].message.content
+        return explanation
+    except Exception as e:
+        if retry_count < MAX_RETRIES:
+            print(f"Error occurred while processing slide {slide_num}: {str(e)}\n")
+            print(f"Retrying to process slide {slide_num}. Retry count: {retry_count + 1}\n")
+            return await process_slide(slide_num, slide_text, retry_count=retry_count + 1)
+        else:
+            raise SlideProcessingError(f"Failed to process slide after {MAX_RETRIES} retries: {str(e)}") from e
+
+
+def extract_slide_text(slide):
+    """
+    Extracts the text content from a slide.
+
+    Args:
+        slide: The slide object.
+
+    Returns:
+        str: The extracted text content of the slide.
+    """
+    slide_text = ""
+    for shape in slide.shapes:
+        if shape.has_text_frame:
+            for paragraph in shape.text_frame.paragraphs:
+                for run in paragraph.runs:
+                    slide_text += run.text
+    return slide_text.strip()
+
+
+async def process_presentation(presentation_path):
+    """
+    Processes a PowerPoint presentation by extracting slide texts and generating explanations for each slide.
+
+    Args:
+        presentation_path (str): The path to the PowerPoint presentation file.
+
+    Returns:
+        list: A list of dictionaries containing the slide number and its corresponding explanation.
+
+    Raises:
+        PresentationProcessingError: If the presentation processing fails.
+    """
+    try:
+        presentation = pptx.Presentation(presentation_path)
+        slides = [extract_slide_text(slide) for slide in presentation.slides]
+
+        explanations = []
+
+        for i, slide_text in enumerate(slides, start=1):
```

Comment:
This doesn't look very synchronic... the whole idea is to let GPT process all the slide in the same time.
For a pptx with 30 slides your code will take minutes to finish.